### PR TITLE
add tooltip to indicator

### PR DESCRIPTION
**Pull request by andirsun**
This PR must to be merged first : https://github.com/elementary/granite/pull/446
close : #74

### CODE REVIEW

**Comment by andirsun** [comment]
## Demo : 
### Enable case : 
on
### Snozed case : 
Screenshot from 2020-12-10 16-52-00

**Review by cassidyjames — CHANGES_REQUESTED** [review]
As we landed in https://github.com/elementary/wingpanel-indicator-keyboard/pull/87, I think the copy here should be,

>Night Light is on

and

>Night Light is off

I suspect the way the markup string is being constructed is also not the most translatable; perhaps it would be better to just have a separate string for each case, like you have with the `accel` string?

**Comment by cassidyjames** [comment]
Something else I'm wondering: should we be handling this markup with a Granite utility to keep its styling in-line with the accel tooltips? It feels bad to be carrying the styling in both Granite and here to match. Thoughts from @elementary/desktop-developers?

**Comment by andirsun** [comment]
> Something else I'm wondering: should we be handling this markup with a Granite utility to keep its styling in-line with the accel tooltips? It feels bad to be carrying the styling in both Granite and here to match. Thoughts from @elementary/desktop-developers?

I have tried 
```
Granite.markup_accel_tooltip ({"<small>Middle-click to snooze</small>"}, _("example"));
```
But the accel doesnt show, I think this is a bug with granite

**Comment by cassidyjames** [comment]
@andirsun no, Granite is working correctly; it is intended to pass keycodes to the accel tooltip, not a string. What I am wondering is if we should have a new utility that does accept a freeform string, like `Granite.markup_tooltip_with_s ()` or something maybe less-terribly named. :smile:

**Comment by andirsun** [comment]
@cassidyjames  Oh I got the idea!, 
Should I work on that change in granite before implementing this ?

File: src/Indicator.vala [hunk 1]
```diff
@@ -88,6 +92,17 @@ public class Nightlight.Indicator : Wingpanel.Indicator {
...
+    private void update_tooltip (bool status) {
+        string context = _("Middle-click to snooze");
+        if (status) {
+            context = _("Middle-click to enable");
+        }
+        indicator_icon.tooltip_markup = Granite.markup_contextual_tooltip (
+            context,
+            _("Night Light is %s".printf (status ? "off" : "on"))
+        );
+    }
```
Comment (cassidyjames) [inline src/Indicator.vala:105]: I think this translates my comments into code:

```suggestion
        string primary_text = _("Night Light is on");
        string secondary_text = _("Middle-click to snooze");

        if (status) {
            primary_text = _("Night Light is snoozed");
            secondary_text = _("Middle-click to enable");
        }

        indicator_icon.tooltip_markup = "%s\n%s".printf (
            primary_text,
            Granite.TOOLTIP_SECONDARY_TEXT_MARKUP.printf (secondary_text)
        );
    }
```

I am a little confused by `status` though; it seems backwards to me? If status is true, I would think Night Light would be on. Perhaps a more clear variable name is warranted.

File: src/Indicator.vala [hunk 2]
```diff
@@ -21,6 +21,7 @@ public class Nightlight.Indicator : Wingpanel.Indicator {
     private Gtk.Spinner? indicator_icon = null;
     private Gtk.StyleContext style_context;
     private Nightlight.Widgets.PopoverWidget? popover_widget = null;
+    private bool nightlight_status;
```
Comment (cassidyjames) [inline src/Indicator.vala:24]:
```suggestion
    private bool snoozed;
```

File: src/Indicator.vala [hunk 3]
```diff
@@ -34,6 +35,7 @@ public class Nightlight.Indicator : Wingpanel.Indicator {
 
     public Indicator () {
         Object (code_name: "wingpanel-indicator-nightlight");
+        nightlight_status = false;
```
Comment (cassidyjames) [inline src/Indicator.vala:38]:
```suggestion
        snoozed = false;
```

File: src/Indicator.vala [hunk 4]
```diff
@@ -71,6 +74,7 @@ public class Nightlight.Indicator : Wingpanel.Indicator {
 
             nightlight_state = !nightlight_manager.snoozed;
             visible = nightlight_manager.active;
+            update_tooltip (nightlight_status);
```
Comment (cassidyjames) [inline src/Indicator.vala:77]:
```suggestion
            update_tooltip (snoozed);
```

File: src/Indicator.vala [hunk 5]
```diff
@@ -88,6 +92,21 @@ public class Nightlight.Indicator : Wingpanel.Indicator {
     public override void opened () {}
 
     public override void closed () {}
+
+    private void update_tooltip (bool status) {
```
Comment (cassidyjames) [inline src/Indicator.vala:96]:
```suggestion
    private void update_tooltip (bool snoozed) {
```

File: src/Indicator.vala [hunk 6]
```diff
@@ -88,6 +92,21 @@ public class Nightlight.Indicator : Wingpanel.Indicator {
...
+        string primary_text = _("Night Light is on");
+        string secondary_text = _("Middle-click to snooze");
+
+        if (status) {
```
Comment (cassidyjames) [inline src/Indicator.vala:100]:
```suggestion
        if (snoozed) {
```

**Review by cassidyjames — CHANGES_REQUESTED** [review]
I think we can change `status` to `snoozed` to be more clear. I think I hit all the instances, but it would be good to double-check.

**Review by cassidyjames — APPROVED** [review]
Huzzah!

File: src/Indicator.vala [hunk 7]
```diff
@@ -71,6 +74,7 @@ public class Nightlight.Indicator : Wingpanel.Indicator {
 
             nightlight_state = !nightlight_manager.snoozed;
             visible = nightlight_manager.active;
+            update_tooltip (snoozed);
```
Comment (danrabbit) [inline src/Indicator.vala:77]: Hold the phone a second. Is this value ever anything other than `false` and is it ever different from `nightlight_manager.snoozed`? Should this be
```suggestion
            update_tooltip (nightlight_manager.snoozed);
```

**Review by danrabbit — COMMENTED** [review]
Hrm, quick question inline

File: src/Indicator.vala (hunk 7 above)
Comment (andirsun) [reply src/Indicator.vala]: @danrabbit  Yep, it works too! Thanks for code review

**Comment by danrabbit** [comment]
Looks good, nice work!

**Pull request merged**